examples/timm-sparseml/main.py

```diff
@@ -0,0 +1,901 @@
+        dataset_eval,
+        input_size=data_config['input_size'],
+        batch_size=args.validation_batch_size_multiplier * args.batch_size,
+        is_training=False,
+        use_prefetcher=args.prefetcher,
+        interpolation=data_config['interpolation'],
+        mean=data_config['mean'],
+        std=data_config['std'],
+        num_workers=args.workers,
+        distributed=args.distributed,
+        crop_pct=data_config['crop_pct'],
+        pin_memory=args.pin_mem,
+    )
+
+    # setup loss function
+    if args.jsd:
+        assert num_aug_splits > 1  # JSD only valid with aug splits set
+        train_loss_fn = JsdCrossEntropy(num_splits=num_aug_splits, smoothing=args.smoothing).cuda()
+    elif mixup_active:
+        # smoothing is handled with mixup target transform
+        train_loss_fn = SoftTargetCrossEntropy().cuda()
+    elif args.smoothing:
+        train_loss_fn = LabelSmoothingCrossEntropy(smoothing=args.smoothing).cuda()
+    else:
+        train_loss_fn = nn.CrossEntropyLoss().cuda()
+    validate_loss_fn = nn.CrossEntropyLoss().cuda()
+
+    # setup checkpoint saver and eval metric tracking
+    eval_metric = args.eval_metric
+    best_metric = None
+    best_epoch = None
+    saver = None
+    output_dir = ''
+    if args.local_rank == 0:
+        output_base = args.output if args.output else './output'
+        exp_name = '-'.join([
+            datetime.now().strftime("%Y%m%d-%H%M%S"),
+            args.model,
+            str(data_config['input_size'][-1])
+        ])
+        output_dir = get_outdir(output_base, 'train', exp_name)
+        decreasing = True if eval_metric == 'loss' else False
+        saver = CheckpointSaver(
+            model=model, optimizer=optimizer, args=args, model_ema=model_ema, amp_scaler=loss_scaler,
+            checkpoint_dir=output_dir, recovery_dir=output_dir, decreasing=decreasing, max_history=args.checkpoint_hist)
+        with open(os.path.join(output_dir, 'args.yaml'), 'w') as f:
+            f.write(args_text)
+
+    ####################################################################################
+    # Start SparseML Integration
+    ####################################################################################
+    sparseml_loggers = (
+        [PythonLogger(), TensorBoardLogger(log_path=output_dir)]
+        if output_dir
+        else None
+    )
+    # determine recipe type to be used if loading from SparseZoo
+    if args.sparseml_recipe_path.startswith("zoo:"):
+        zoo_recipe_type = "transfer" if args.sparse_transfer_learn else "original"
+    else:
+        zoo_recipe_type = None
+    manager = ScheduledModifierManager.from_yaml(
+        args.sparseml_recipe_path, zoo_recipe_type=zoo_recipe_type
+    )
+    optimizer = ScheduledOptimizer(
+        optimizer,
+        model,
+        manager,
+        steps_per_epoch=len(loader_train),
+        loggers=sparseml_loggers
+    )
+    start_epoch = manager.min_epochs  # override min_epochs
```

Comment:
I'm not sure if users would expect this. For example, if we train for an epoch without doing anything to modify the process and then start pruning, this would improperly start at 1 and not 0 which the user had intended.

Additionally, we would like the user to be able to start in the middle of a recipe when restoring from a checkpoint

Reply:
good point, removing